### PR TITLE
.travis.yml: Build on xenial using template

### PR DESCRIPTION
**Pull request by jayvdb**
https://github.com/coala/coala/issues/6053 is only
occurring in the Travis trusty environment.
Upgrading to xenial bypasses that problem, but
requires a few workarounds, especially replacing the
Python 3.4.2 job with Python '3.4' which causes
Travis to provide Python 3.4.8.

Add gravis to template to switch jdk versions on Xenial.

Uninstall setuptools multiple times on Xenial for the same
reason it was required once on Trusty.  Also remove
unnecessary pipenv which causes pip to complain about
incompatibilities.

Disable new template for JinjaBear due to its inability
to handle indentation.

.misc/check_moban.sh is no longer needed as the
Travis template provides a moban job, and the latest
moban handles branches.

Related to https://github.com/coala/coala/issues/4708
Related to https://github.com/coala/coala/issues/6053

Implements https://gitlab.com/coala/mobans/merge_requests/135 , which needs to be approved first.

### CODE REVIEW

File: .moban.dt/core-travis.yml.jj2 [hunk 1]
```diff
@@ -0,0 +1,151 @@
+{% extends 'ci/travis.yml.jj2' %}
+
+{% block custom_python_versions %}
+python:
+{% if travis_dist not in ['precise', 'trusty'] %}
```
Comment (jayvdb) [inline .moban.dt/core-travis.yml.jj2]: Note that if we find a better solution for https://github.com/coala/coala/issues/6053 on trusty, the old Python 3.4.2 minimum dep will be restored, or maybe before then the need for py34 has evaporated.

Comment (jayvdb) [reply .moban.dt/core-travis.yml.jj2]: Template moved to https://gitlab.com/coala/mobans/merge_requests/135 , as coala-quickstart and others need the same fixes

File: .coafile [hunk 2]
```diff
@@ -134,6 +134,8 @@ ignore = node_modules/
 [jinja2]
 bears = Jinja2Bear
 files = **.jj2
+check_end_labels = False
+ignore = .moban.dt/core-travis.yml.jj2
```
Comment (KVGarg) [inline .coafile]: is this required? As you have moved this file

Comment (jayvdb) [reply .coafile]: Good catch.  That should be removed.

File: .travis.yml [hunk 3]
```diff
@@ -14,19 +16,27 @@ stages:
...
 
-.check_moban: &check_moban
+.moban: &moban
   <<: *disable_global
   python: 3.6
   stage: moban
-  install: pip install moban
-  script: .misc/check_moban.sh
-  if: branch != master OR type = pull_request
+  install: pip install moban>=0.0.4
```
Comment (KVGarg) [inline .travis.yml]: Just a thought, In [test-requirements.txt](https://github.com/coala/coala/blob/master/test-requirements.txt#L15) we're installing another version of `moban` for python version > 3.0 and here we're installing a different version. Shouldn't they be same?

**Pull request merged**